iotics/lib/identity/register/document_builder.py

```diff
@@ -28,7 +28,11 @@ def __init__(self):
 
     def _check_and_update_map(self, name: str, val: object, key_map: dict, message_prefix: str):
         if name in self.name_set:
-            raise IdentityRegisterDocumentKeyConflictError(f'{message_prefix} \'{name}\' already in use')
+            if isinstance(val, RegisterDelegationProof) and key_map[name].controller == val.controller:
```

Comment:
What are we trying to achieve here?

Reply:
I found that if you call `create_twin_with_control_delegation` twice it would raise this error

Reply:
twice with the same name?
Maybe  we could improve the check so it raises only if the new key is different from the existing one but keep it generic:

```
if name in self.name_set and key_map.get(name) != val:
```

Reply:
Cool thanks I'll update the check